<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="arm">
  <xacro:include filename="$(find arm)/urdf/arm.urdf" />

  <!-- ros2_control configuration -->
  <ros2_control name="arm" type="system">
    <hardware>
      <plugin>hardware/SimHardware</plugin>
    </hardware>

    <joint name="joint_1">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <param name="pin">0</param>
      <param name="min_pwm_width_microseconds">500</param>
      <param name="max_pwm_width_microseconds">2500</param>
      <param name="min_position_radians">0</param>
      <param name="max_position_radians">3.1416</param>
      <param name="starting_position_radians">1.5708</param>
    </joint>
    <joint name="joint_2">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <param name="pin">0</param>
      <param name="min_pwm_width_microseconds">500</param>
      <param name="max_pwm_width_microseconds">2500</param>
      <param name="min_position_radians">0</param>
      <param name="max_position_radians">3.1416</param>
      <param name="starting_position_radians">1.5708</param>
    </joint>
    <joint name="joint_3">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <param name="pin">0</param>
      <param name="min_pwm_width_microseconds">500</param>
      <param name="max_pwm_width_microseconds">2500</param>
      <param name="min_position_radians">0</param>
      <param name="max_position_radians">3.1416</param>
      <param name="starting_position_radians">1.5708</param>
    </joint>
    <joint name="joint_4">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <param name="pin">0</param>
      <param name="min_pwm_width_microseconds">500</param>
      <param name="max_pwm_width_microseconds">2500</param>
      <param name="min_position_radians">0</param>
      <param name="max_position_radians">3.1416</param>
      <param name="starting_position_radians">1.5708</param>
    </joint>
    <joint name="joint_5">
      <command_interface name="position"/>
      <state_interface name="position"/>
      <param name="pin">0</param>
      <param name="min_pwm_width_microseconds">500</param>
      <param name="max_pwm_width_microseconds">2500</param>
      <param name="min_position_radians">0</param>
      <param name="max_position_radians">3.1416</param>
      <param name="starting_position_radians">1.5708</param>
    </joint>
  </ros2_control>
</robot>
